CodingStandard.md

```diff
@@ -1,34 +1,120 @@
+Style Guide, found here: <https://google.github.io/styleguide/cppguide.html>
+
+If a topic is not covered in any of these documents, but it seems that it needs to be addressed, this document may (and
+should) be modified to reflect the most up-to-date standard employed by the Trickfire Robotics Software Team.
+
+General Standards
+=======================================================================================================================
+
+##Naming
+All names should be descriptive of their purpose. Prefer long, descriptive names to short, confusing names.
+
+**Files**
+
+- Source files should be `lower_case`. C++ files should use either the `.h` or `.cpp` file extensions.
+
+- In general, the use of extra prefixes for file names is discouraged:
+  - Source files representing ROS Nodes should not contain a `node_` prefix or `node_` suffix.
+  - Message (`.msg`), Service (`.srv`), and Action (`.action`) files should be `CamelCase`, not contain the respective
+  "Message", "Service", or "Action" identifiers in the name.
+  - Configuration (`.yaml`) files should not contain `_config` or `_param` suffixes.
+  - Launch files (`.launch`) should not contain the `_launch` suffix.
+- Exeptions:
+  - Packages should be `lower_case` and be contain the `tfr_` prefix (as package names are global to the entire ROS
+    ecosystem).
+  - Files defining an abstract base class should contain a `_interface` suffix.
+
+**Identifiers**
+
+C++ code should adhere to the naming conventions below to promote consistency
+throughout the project.
+```
+  - Pre-Processor Macros: `UPPER_CASE`
+  - Class, Struct, Union, and Enum Types: `CamelCase`
+  - Functions and Class Methods: `lower_case`
+  - Compile-time constants (const, constexpr, enumerators, etc.): `kCamelCase`
+  - Variables and Function Parameters: `lower_case`
+  - constexpr Methods and Functions: `lower_case`
+  - Non-const struct member variables: `lower_case`
+  - Non-const class member variables: `m_lower_case`
+  - Const member variables: `kCamelCase`
+  - Template type parameters: `CamelCase`
+  - Template value parameters: `kCamelCase`
+  - Namespaces: `lower_case`
 
-During a code review, reviewers may wish to consult this page to help determine
-if the source code aligns with the project coding standards, guidelines, and
-conventions.
+- Abstract base classes (interfaces) should be prefixed with I, as in
+  `IContainer`.
 
-If a scenario is not covered on this page, use good judgment and remember
-*consistency is king*.  If there is an established way of doing something in
-other parts of the code base, that's probably the way it should be done.
+- Acronyms and abbreviations should be avoided, unless they are well-known and
+  commonly encountered throughout the project.
+```
+##Formatting
+
+- All C++ and Python code should use 4 spaces instead of tabs.
+- C++ braces should always be placed on their own line.
```

Comment:
Hey @RyanBerge @AdamuKaapan there is an edge case that I want thrown in here. Uniform braced initialization. I use it, it's use is encouraged by scott meyers, herb sutter, the ISO c++ standard guidelines, and bjarne strostrup. 

It avoids narrowing and widening errors, and eliminates the c++ most vexing parse.

E.G:  
Server server{ n, "do_action", boost::bind(&execute, _1, &server), false };

It would be silly to put on multiple lines. I want a clause allowing for uniform braced initialization.

Reply:
I support this. I had to look up what it was first, I've never used those before, but that's a perfectly reasonable clause to add it seems like.

Reply:
Sure. I don't think anyone intended to restrict brace-initialization to multiple lines.